source/client/stream_decoder.cc

```diff
@@ -57,7 +57,15 @@ void StreamDecoder::onPoolFailure(Envoy::Http::ConnectionPool::PoolFailureReason
 
 void StreamDecoder::onPoolReady(Envoy::Http::StreamEncoder& encoder,
                                 Envoy::Upstream::HostDescriptionConstSharedPtr) {
-  encoder.encodeHeaders(request_headers_, true);
+  encoder.encodeHeaders(request_headers_, request_body_size_ == 0);
+  if (request_body_size_ > 0) {
+    // TODO(oschaaf): We can probably get away without allocating/copying here if we pregenerate
+    // potential request-body candidates up front. Revisit this when we have non-uniform request
+    // distributions and on-the-fly reconfiguration in place.
+    std::string body(request_body_size_, 'a');
```

Comment:
One thing to keep in mind is that repeated `a` will compress very easily if we ever have compression on the data plane.